### ISSUE TITLE
Error running sqlfluff lint with templater = dbt

### ISSUE THREAD

**Issue by angelcontreras-zebra** [sqlfluff/sqlfluff#844]
## Error

anyone tried to run use `templater = dbt` in the `.sqlfluff` file and got this error?

```
Unable to lint models/model.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/linter.py", line 1259, in lint_path
    self.lint_string(
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/linter.py", line 1052, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/linter.py", line 793, in parse_string
    templated_file, templater_violations = self.templater.process(
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/templaters/dbt.py", line 188, in process
    return self._unsafe_process(fname, in_str, config)
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/templaters/dbt.py", line 222, in _unsafe_process
    selected = self.dbt_selector_method.search(
  File "/usr/local/lib/python3.9/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/templaters/dbt.py", line 107, in dbt_selector_method
    if "0.17" in self.dbt_version:
  File "/usr/local/lib/python3.9/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.9/site-packages/sqlfluff/core/templaters/dbt.py", line 42, in dbt_version
    from dbt.version import get_installed_version
  File "/usr/local/lib/python3.9/site-packages/dbt/version.py", line 100, in <module>
    installed = get_installed_version()
  File "/usr/local/lib/python3.9/site-packages/dbt/version.py", line 29, in get_installed_version
    return dbt.semver.VersionSpecifier.from_version_string(__version__)
  File "/usr/local/lib/python3.9/site-packages/dbt/semver.py", line 100, in from_version_string
    return cls.from_dict(matched)
  File "/usr/local/lib/python3.9/site-packages/hologram/__init__.py", line 618, in from_dict
    cls.validate(data)
  File "/usr/local/lib/python3.9/site-packages/hologram/__init__.py", line 957, in validate
    schema = _validate_schema(cls)
  File "/usr/local/lib/python3.9/site-packages/hologram/__init__.py", line 183, in _validate_schema
    schema = schema_cls.json_schema()
  File "/usr/local/lib/python3.9/site-packages/hologram/__init__.py", line 942, in json_schema
    return cls._json_schema_recursive(
  File "/usr/local/lib/python3.9/site-packages/hologram/__init__.py", line 913, in _json_schema_recursive
    schema = cls._schema_defs_from_cache(definitions)
  File "/usr/local/lib/python3.9/site-packages/hologram/__init__.py", line 900, in _schema_defs_from_cache
    schema = cls._collect_json_schema(definitions)
  File "/usr/local/lib/python3.9/site-packages/hologram/__init__.py", line 863, in _collect_json_schema
    properties[target_field], is_required = cls._get_field_schema(
  File "/usr/local/lib/python3.9/site-packages/hologram/__init__.py", line 809, in _get_field_schema
    field_schema, required = cls._get_schema_for_type(
  File "/usr/local/lib/python3.9/site-packages/hologram/__init__.py", line 783, in _get_schema_for_type
    raise ValidationError(f"Unable to create schema for '{type_name}'")
hologram.ValidationError: Unable to create schema for 'Optional'
```

## Expected Behaviour
show the normal ident error that i need to fix.

## Observed Behaviour
Only if i use: `templater = dbt` in the .sqlfluff file.

## Steps to Reproduce
Witht the specified Configuration just need to run the below code and you will get the error:
```
echo "  SELECT a  +  b FROM tbl;  " > test.sql
sqlfluff lint test.sql
```

## Version
sqlfluff, version 0.4.1
Python 3.8.2 (pip 21.0.1)
Python 2.7.16

## Configuration
```
[sqlfluff]
verbose = 0
templater = dbt
dialect = ansi

# dbt templating does not keep trailing new lines (L009)
exclude_rules = L009

[sqlfluff:templater:jinja]
load_macros_from_path=macros
```

**Comment by NiallRees** [comment]
Hi @angelcontreras-zebra your stacktrace shows that you're using python 3.9 - dbt does not currently support 3.9. Going to close this for now.
